src/coreclr/debug/runtimeinfo/datadescriptor.h

```diff
@@ -107,9 +107,18 @@ CDAC_TYPE_BEGIN(Thread)
 CDAC_TYPE_INDETERMINATE(Thread)
 CDAC_TYPE_FIELD(Thread, /*uint32*/, Id, cdac_offsets<Thread>::Id)
 CDAC_TYPE_FIELD(Thread, /*nuint*/, OSId, cdac_offsets<Thread>::OSId)
+CDAC_TYPE_FIELD(Thread, /*uint32*/, State, cdac_offsets<Thread>::State)
+CDAC_TYPE_FIELD(Thread, /*uint32*/, PreemptiveGCDisabled, cdac_offsets<Thread>::PreemptiveGCDisabled)
```

Comment:
I would reconsider this. This is obstensibly a `BOOL` value, but that is hard to version. I would instead treat this as a set of flags with only the lowest order bit defined. In fact, that goes for any value in the cDAC, boolean values should generally be avoided.

Reply:
> Are you saying we should do a & 0x1 first instead of just checking != 0 when converting the value to a boolean?

Yes for this and yes for the follow up about having boolean values in the contract. Encoding a boolean in a data contract is a bad idea because it locks us into ignoring higher order bits or worse, needing to interpret any bit pattern as "true". Since the name itself is merely convenience checking a specific bit and assuming others are ignored for now is the best way to avoid getting ourselves into a bad situation.

The general philosophy for any sort of data contract is that we should be looking at all bits as significant. One can see this for types likes integral types such as `int` or floating point types such as `float`. Even pointers should be interpreted as "all bits are equally important". Boolean values are suspect though because none of the languages we consume work in terms of bits - which a boolean is literally just that. This means avoiding contract types that encode concepts smaller than a byte, boolean is the worst offender of this.